pkg/bgpv1/manager/reconcilerv2/service.go

```diff
@@ -302,6 +320,70 @@ func (r *ServiceReconciler) getDesiredSvcRoutePolicies(p ReconcileParams, desire
 	return desiredSvcRoutePolicies, nil
 }
 
+// handleOverlappingDesiredSvcRoutePolicies handles the case of a desired BGP advertisement that overlaps a previous
+// selector match. Overlaps are handled as additive operations for attributes that can be unioned (standard and large
+// communities). If local preference was set, it remains set to the first advertisement matched.
+//
+// This function modifies existingPolicy to contain the unions described above.
+func (r *ServiceReconciler) handleOverlappingDesiredSvcRoutePolicies(
+	existingPolicy *types.RoutePolicy,
+	candidatePolicy *types.RoutePolicy,
+	advert v2alpha1.BGPAdvertisement,
+) {
+
+	// For tests, we must maintain the original order.  Communities will hold
+	// the unique set of standard communities defined across both policies.
+	communities := []string{}
+	communitiesSeen := sets.New[string]()
+
+	largeCommunities := []string{}
+	largeCommunitiesSeen := sets.New[string]()
+
+	// Extract the existing policy's communities
+	for _, statement := range existingPolicy.Statements {
+		communities = append(communities, statement.Actions.AddCommunities...)
+		communitiesSeen.Insert(statement.Actions.AddCommunities...)
+
+		largeCommunities = append(largeCommunities, statement.Actions.AddLargeCommunities...)
+		largeCommunitiesSeen.Insert(statement.Actions.AddLargeCommunities...)
+	}
+	logMessagePreviousCommunities := fmt.Sprint(communities)
+	logMessagePreviousLargeCommunities := fmt.Sprint(largeCommunities)
+
+	// Extract this policy's communities
+	for _, statement := range candidatePolicy.Statements {
+		for _, community := range statement.Actions.AddCommunities {
+			if communitiesSeen.Has(community) {
+				continue
+			}
+			communities = append(communities, community)
+		}
+
+		for _, largeCommunity := range statement.Actions.AddLargeCommunities {
+			if largeCommunitiesSeen.Has(largeCommunity) {
+				continue
+			}
+			largeCommunities = append(largeCommunities, largeCommunity)
+		}
+	}
+
+	// Update the existing policy's communities to be the updated sequence created above
+	for _, statement := range existingPolicy.Statements {
```

Comment:
It seems all statements will get same communities, don't we need to match on the action? 

For eg, Policy A
Statement 1, Cond 1, Action 1a, 2a 
Statement 2, Cond 2, Action 3a

Policy B
Statement1, Cond 1, Action 1b

Expected Merged Policy
Statement1: Cond 1, Action 1a, 2a, 1b
Statement2: Cond 2, Action 3a

In the code, we will now get the following
Merged Policy
Statement 1: Cond 1, Action 1a, 2a, 3a, 1b 
Statement 2: Cond 2, Action 1a, 2a, 3a, 1b

Reply:
Thanks @harsimran-pabla for the review!  I'll make the requested changes and be back in touch.

Re: "It seems all statements will get the same communities",

It looks to me that they all get the same communities.  Help me double-check this:
Looking at `master`:

1) `getDesiredSvcRoutePolicies()` for the ClusterIP case calls `getClusterIPRoutePolicy()`

2) `getClusterIPRoutePolicy()` then calls `CreatePolicy()` toward the end

3) `CreatePolicy()` calls `getCommunities()` and passes the original `advert` variable in.  This variable comes from the for loop on `getDesiredSvcRoutePolicies()`

Reply:
In the current implementation, you're right - callers always create policy with single statement ( we create different policy for v4 and v6 ). However, `merge` method is assuming that multiple matched advertisements will have same conditions ( which is true today ) but may not be always the case. We can tackle this when such case arises. 

So, we can assume now that if policy name matches, statements and conditions will be same. Only actions of the statements will be merged.

For eg, this is the only case

Policy A
Statement 1, Cond 1, Action 1a, 2a
Policy B
Statement1, Cond 1, Action 1b

Expected Merged Policy
Statement1: Cond 1, Action 1a, 2a, 1b